processor/README.md

```diff
@@ -254,22 +289,101 @@ Refer to [config.yaml](batchprocessor/testdata/config.yaml) for detailed
 examples on using the processor.
 
 
-## <a name="queued_retry"></a>Queued Retry Processor
+## <a name="memory-limiter"></a>Memory Limiter Processor
+
+The memory_limiter processor is used to prevent out of memory situations on
+the collector. Given that the amount and type of data a collector processes is
+environment specific and resource utilization of the collector is also dependent
+on the configured processors, it is important to put checks in place regarding
+memory usage. The memory_limiter processor offers the follow safeguards:
+
+- Ability to define an interval when memory usage will be checked and if memory
+usage exceeds a defined limit will trigger GC to reduce memory consumption.
+- Ability to define an interval when memory usage will be compared against the
+previous interval's value and if the delta exceeds a defined limit will trigger
+GC to reduce memory consumption.
+
+In addition, there is a command line option (`mem-ballast-size-mib`) which can be
+used to define a ballast, which allocates memory and provides stability to the
+heap. If defined, the ballast increases the base size of the heap so that GC
+triggers are delayed and the number of GC cycles over time is reduced. While the
+ballast is configured via the command line, today the same value configured on the
+command line must also be defined in the memory_limiter processor.
+
+Note that while these configuration options can help mitigate out of memory
+situations, they are not a replacement for properly sizing and configuring the
+collector. For example, if the limit or spike thresholds are crossed, the triggered
+GC may result in dropped data.
```

Comment:
nit: I find the suggestion below a bit more optimistic :) - btw, multiline suggestion, yay!
```suggestion
collector. For example, if the limit or spike thresholds are crossed, the collector
will return errors to all receive operations until enough memory is freed.
```